services/identity-service/src/did/did.service.ts

```diff
@@ -8,21 +8,43 @@ import { VaultService } from '../utils/vault.service';
 import { Identity } from '@prisma/client';
 @Injectable()
 export class DidService {
+  static getKeySignType = (algo: string): any => {
+    switch (algo) {
+      case 'Ed25519':
+      case 'EdDSA':
+        return {
+          keyType: "JsonWebKey2020",
+          signType: "JsonWebSignature2020"
+        };
+      case 'secp256k1':
+      case 'ES256K':
+        return {
+          keyType: "EcdsaSecp256k1VerificationKey2019",
+          signType: "EcdsaSecp256k1Signature2019"
+        };
+      default:
+        return {};
+    }
+  }
+
   constructor(private prisma: PrismaService, private vault: VaultService) {}
 
   async generateDID(doc: GenerateDidDTO): Promise<DIDDocument> {
     // Create private/public key pair
     let authnKeys;
+    let signingAlgorithm: string = process.env.SIGNING_ALGORITHM;
     try {
-      authnKeys = await ION.generateKeyPair(process.env.SIGNING_ALGORITHM as string);
-    } catch (err) {
+      authnKeys = await ION.generateKeyPair(signingAlgorithm);
```

Comment:
Is this throwing an error when we enter wrong AlgoType?

Reply:
Yes